syntax = "proto3";


service KeyValue {
    rpc Put (PutRequest) returns (Response) {}
    rpc Get (GetRequest) returns (Response) {}
    rpc GetAll (All) returns (ResponseAll) {}
}

message PutRequest {
    string key = 1;
    string value = 2;
}

message GetRequest {
    string key = 1;
}

message All {}

message Response {
    string message = 1;
}

message ResponseAll {
    repeated string message = 1;
}